### DISCUSSION TITLE
Headless templates not found even with -tags specified

### DISCUSSION THREAD

**Discussion by Techbrunch** [projectdiscovery/nuclei#715] [Q&A] [answered]
**Describe the bug**

For some reasons nuclei is not able to find the headless templates

```
nuclei -target https://target.com -t /Users/xxx/nuclei-templates -tags headless

                       __     _
     ____  __  _______/ /__  (_)
    / __ \/ / / / ___/ / _ \/ /
   / / / / /_/ / /__/ /  __/ /
  /_/ /_/\__,_/\___/_/\___/_/   v2.3.7

		projectdiscovery.io

[INF] Loading templates...
[INF] Loading workflows...
[FTL] Error, no templates were found.
```

But the templates are clearly there:
```
pwd
/Users/xxx/nuclei-templates
ls headless
.rwxr-xr-x 1.0k xxx 17 May 11:00 dvwa-headless-automatic-login.yaml
.rwxr-xr-x 1.8k xxx 17 May 11:00 postmessage-tracker.yaml
.rwxr-xr-x 3.2k xxx 17 May 11:00 prototype-pollution-check.yaml
.rwxr-xr-x 2.6k xxx 17 May 11:00 window-name-domxss.yaml
```

**Nuclei version**

```
v2.3.7
```

**Comment by geeknik** [comment, marked as answer]
Hi, you must also add the `-headless` flag to your command line. 🤙

**Reply by Techbrunch** [reply]
Ok thanks for the info, I think there might be something to improve there in terms of UI.

**Reply by ehsandeep** [reply]
Yes, looks like Error details can be more verbose to include exact error details.

**Reply by Techbrunch** [reply]
Well I think more than just details. If `-headless` is required to run headless templates why would it be part of the default ignored templates ? If you specify the flag it means that you want to run headless templates right ?

**Reply by ehsandeep** [reply]
Thanks for adding more points to this, [updated the default ignore list](https://github.com/projectdiscovery/nuclei-templates/commit/1ab9cce6192a9efe4b244bc419f93ac2a3df5b92) to remove headless-based template, now it should be usable with `-headless` flag only.

We were being extra careful about someone accidentally running headless templates on tons of URLs causing performance issue as it makes use of browser based engine instead of standard go `net/http`
